server/store/views.py

```diff
@@ -47,6 +47,23 @@ class ProductPurchaseException(APIException):
     }
 
 
+def log_error(error_name, error_message, product):
```

Comment:
discussion: This logging helper seems a bit specialized, but I think that keeps it approachable to read and infer more abstraction.